tsl/src/continuous_aggs/invalidation_threshold.c

```diff
@@ -169,7 +183,7 @@ invalidation_threshold_set_or_get(const ContinuousAgg *cagg,
 		.data = &updatectx,
 		.limit = 1,
 		.tuple_found = invalidation_threshold_scan_update,
-		.lockmode = RowExclusiveLock,
+		.lockmode = ShareUpdateExclusiveLock,
```

Comment:
We take the same lock in `invalidation_threshold_initialize` when the threshold is initialized. Conflicting with this lock should be fine since the threshold should be initialized when we try to update it. Reader (e.g., `get_lowest_invalidated_time_for_hypertable` and `cache_inval_htab_write` are using an `AccessShareLock`. So, we should not conflict with them and reads should be still possible.).